tools/clang/tools/libclang/dxcisenseimpl.cpp

```diff
@@ -1823,6 +1823,36 @@ HRESULT DxcTranslationUnit::GetInclusionList(unsigned *pResultCount,
   return S_OK;
 }
 
+_Use_decl_annotations_
+HRESULT DxcTranslationUnit::CodeCompleteAt(
+	char *fileName, unsigned line, unsigned column,
+	IDxcUnsavedFile **pUnsavedFiles, unsigned numUnsavedFiles,
+	DxcCodeCompleteFlags options, IDxcCodeCompleteResults **pResult)
+{
+  if (pResult == nullptr) return E_POINTER;
+
+  DxcThreadMalloc TM(m_pMalloc);
+
+  CXUnsavedFile *files;
+  HRESULT hr = SetupUnsavedFiles(pUnsavedFiles, numUnsavedFiles, &files);
+  if (FAILED(hr))
+    return hr;
+
+  CXCodeCompleteResults *results = clang_codeCompleteAt(
+      m_tu, fileName, line, column, files, numUnsavedFiles, options);
+
+  if (results == nullptr) return E_FAIL;
+  *pResult = nullptr;
+  DxcCodeCompleteResults *newValue =
+      new (std::nothrow) DxcCodeCompleteResults();
+  if (newValue == nullptr)
+    return E_OUTOFMEMORY;
```

Comment:
(Minor) If you get here, you leak `results`. You might be able to use a std::unique_ptr<CXCodeCompleteResults, clang_disposeCodeCompleteResults> ?

Reply:
Good point Tristan.  We do care about leaks, even though this one is very unlikely.  If it were tested under the OOM test, it would complain.

Reply:
Oh and we also need `CleanupUnsavedFiles`. Our intellisense interfaces are very error-prone in terms of memory management :(

Reply:
Filed #2234

Reply:
Thanks, good catch - I've fixed both of these.